exercises/largest-series-product/largest-series-product.example.ts

```diff
@@ -0,0 +1,52 @@
+export default class Series {
+  private numberString: string
+  private digits: number[]
+
+  constructor(numberString: string) {
+    if (numberString.match('[^0-9]')) {
+      throw new Error('Invalid input.')
+    }
+
+    this.numberString = numberString
+    this.digits = this.getDigits()
+  }
+
+  getDigits() {
+    return [...this.numberString].map((digit) => parseInt(digit, 10))
+  }
+
+  largestProduct(size: number) {
+    if (size < 0) {
+      throw new Error('Invalid input.')
+    }
+
+    let product,
+      max = 0
+    this.slices(size).forEach((slice) => {
```

Comment:
What are your thoughts on utilizing `map` instead of `forEach` to produce a list of values and then taking the maximum value out of that list (with a default of zero)?